### PR TITLE
cargo.eclass: Use a regex to fix crate name/version extraction

### PR DESCRIPTION
**Pull request by candrews**
Closes: https://bugs.gentoo.org/705044
Signed-off-by: Craig Andrews <candrews@gentoo.org>

### CODE REVIEW

**Comment by candrews** [comment]
> I would double check a pre version as well.

Tested with `rand-0.7.0-pre.2` and it worked correctly. If you have any others you'd like to have tested, please let me know.

**Comment by gyakovlev** [comment]
ultimate test will be emerging all cargo.eclass consumers =)

File: eclass/cargo.eclass [hunk 1]
```diff
@@ -39,16 +39,13 @@ ECARGO_VENDOR="${ECARGO_HOME}/gentoo"
 # @DESCRIPTION:
 # Generates the URIs to put in SRC_URI to help fetch dependencies.
 cargo_crate_uris() {
+	readonly regex='^(.*)-([0-9]+\.[0-9]+\.[0-9]+.*)$'
```
Comment (gyakovlev) [inline eclass/cargo.eclass]: make it
```sh
local -r regex='^(.*)-([0-9]+\.[0-9]+\.[0-9]+.*)$'
```

that way it's both local and readonly.

**Comment by gyakovlev** [comment]
if someone does not follow semantic versioning it will fail, because it does not match, for example `foo-1.0` .
cargo guide suggests using versions with three numeric parts, but I'm not sure if it enforces it.
otherwise looks ok.

**Comment by gyakovlev** [comment]
did some silly local tests, looks like it does enforce it.
here's the library it uses https://github.com/steveklabnik/semver-parser/blob/master/src/version.rs
regex seems to be correctly working with most tescases from that file that make sense for us.

going to do some builds with it.

**Comment by gyakovlev** [comment]
ok I ran repoman manifest on `CRATES` consumers, no changes to manifests.
merged, thanks!

**Pull request closed without merging**